lib/reservation_tracker.rb

```diff
@@ -0,0 +1,116 @@
+require_relative "reservation"
+require_relative "hotel_block"
+require "date"
+require "pry"
+
+class ReservationTracker
+  attr_reader :total_room
+  attr_accessor :reservations, :reservations_per_room, :block_reservations
+
+  def initialize
+    @total_room = (1..20).to_a
+    @reservations = []
+    @reservations_per_room = {}
+    @block_reservations = {}
+  end
+
+  def reservations_by_date(date)
+    res_date = Date.parse(date)
+    reservations_at_date = @reservations.select { |reservation| reservation.start_date == res_date }
+    return reservations_at_date
+  end
+
+  def is_overlap?(start_date, end_date, reservation)
+    return reservation.end_date > start_date && reservation.start_date < end_date
+  end
+
+  def same_daterange?(start_date, end_date, reservation)
```

Comment:
I have moved this method to Reservation class and Hotel Block class. I think this method is needed in this program since it is checking wether a reservation of a room in a hotel block has the same start_date and end_date with the hotel block for that reservation to be made and added to the list of reservations.